### DISCUSSION TITLE
HWA Failed for playing movies. Are there anyone having any idea, THX a lot!

### DISCUSSION THREAD

**Discussion by zhongjiongsheng** [jellyfin/jellyfin#7462] [Q&A]
My device is a Qnap TS-462c, which has a cpu of N4505. And the OS is QTS 5.0.
I used a docker to pull the lastest version of Jellyfin, which is 10.8.0 from nyanmisaka/jellyfin @nyanmisaka. 
And if I turned off the HWA (VAAPI or QSV) , everything is OK, movies could be played by software transcoding, while they were always
stumble to watch. Then I tried both VAAPI and QSV, none of them worked for me.
I am sure of the permission of '/dev/dri' is OK. It's a 777 now. And the path was mounted  by:
`  --device /dev/dri:/dev/dri \
  --device /dev/dri/renderD128:/dev/dri/renderD128 \
  --device /dev/dri/card0:/dev/dri/card0 \`
The logs that I would not play movies successfully are listed as follow:
`ffmpeg version 4.4.1-Jellyfin Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10 (Debian 10.2.1-6)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-shared --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-static --enable-gmp --enable-gnutls --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --arch=amd64 --enable-opencl --enable-vaapi --enable-amf --enable-libmfx --enable-vdpau --enable-ffnvcodec --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'file:/data/movies/你的名字。 (2016)/你的名字。 (2016) 2160p AAC.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    encoder         : Lavf58.20.100
  Duration: 01:46:37.65, start: 0.000000, bitrate: 5231 kb/s
  Stream #0:0(und): Video: hevc (Main) (hev1 / 0x31766568), yuv420p(tv), 3840x2160 [SAR 1:1 DAR 16:9], 5164 kb/s, 25 fps, 25 tbr, 12800 tbn, 25 tbc (default)
    Metadata:
      handler_name    : VideoHandler
      vendor_id       : [0][0][0][0]
  Stream #0:1(jpn): Audio: aac (HE-AAC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 62 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
Stream mapping:
  Stream #0:0 -> #0:0 (hevc (native) -> h264 (h264_vaapi))
  Stream #0:1 -> #0:1 (aac (native) -> aac (native))
Press [q] to stop, [?] for help
[h264_vaapi @ 0x55571c6a2200] Driver does not support VBR RC mode (supported modes: CQP).
Error initializing output stream 0:0 -- Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
[aac @ 0x55571c750140] Qavg: 50571.980
[aac @ 0x55571c750140] 2 frames left in the queue on closing
Conversion failed!`

Another Error when I tried to play "Thor" looked like:
`Stream mapping:
  Stream #0:0 -> #0:0 (hevc (native) -> h264 (h264_vaapi))
  Stream #0:1 -> #0:1 (truehd (native) -> aac (native))
Press [q] to stop, [?] for help
[hevc @ 0x557fca2c34c0] Skipping NAL unit 62
[hevc @ 0x557fca2d6ac0] Skipping NAL unit 62
[hevc @ 0x557fca2df080] Skipping NAL unit 62
[hevc @ 0x557fca2c34c0] Skipping NAL unit 62
[Parsed_tonemap_vaapi_1 @ 0x557fca387b40] Failed to start picture processing: 1 (operation failed).
Error while filtering: Input/output error
Failed to inject frame into filter network: Input/output error
Error while processing the decoded data for stream #0:0
[aac @ 0x557fca2c2bc0] Qavg: 59050.461
[aac @ 0x557fca2c2bc0] 2 frames left in the queue on closing
Conversion failed!`

Would someone have any experience to solve these problem. THX A LOT!

**Comment by nyanmisaka** [comment]
> [Parsed_tonemap_vaapi_1 @ 0x557fca387b40] Failed to start picture processing: 1 (operation failed). 

VPP tonemapping is not supported by N4505. Please unckeck it and use OpenCL tonemapping.

**Reply by zhongjiongsheng** [reply]
Answer comes up in a light speed. Thank you!!! Let me try it now.

**Comment by zhongjiongsheng** [comment]
### Sorry for the format issue. It looks mixed up. 

#### when playing Thor, errors looked like:
Stream mapping:
  Stream #0:0 -> #0:0 (hevc (native) -> h264 (h264_vaapi))
  Stream #0:1 -> #0:1 (truehd (native) -> aac (native))
Press [q] to stop, [?] for help
[hevc @ 0x557fca2c34c0] Skipping NAL unit 62
[hevc @ 0x557fca2d6ac0] Skipping NAL unit 62
[hevc @ 0x557fca2df080] Skipping NAL unit 62
[hevc @ 0x557fca2c34c0] Skipping NAL unit 62
[Parsed_tonemap_vaapi_1 @ 0x557fca387b40] Failed to start picture processing: 1 (operation failed).
Error while filtering: Input/output error
Failed to inject frame into filter network: Input/output error
Error while processing the decoded data for stream #0:0
[aac @ 0x557fca2c2bc0] Qavg: 59050.461
[aac @ 0x557fca2c2bc0] 2 frames left in the queue on closing
Conversion failed!
*********************************************************************

#### when playing "你的名字。", errors looked like:
ffmpeg version 4.4.1-Jellyfin Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10 (Debian 10.2.1-6)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-shared --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-static --enable-gmp --enable-gnutls --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --arch=amd64 --enable-opencl --enable-vaapi --enable-amf --enable-libmfx --enable-vdpau --enable-ffnvcodec --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'file:/data/movies/你的名字。 (2016)/你的名字。 (2016) 2160p AAC.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    encoder         : Lavf58.20.100
  Duration: 01:46:37.65, start: 0.000000, bitrate: 5231 kb/s
  Stream #0:0(und): Video: hevc (Main) (hev1 / 0x31766568), yuv420p(tv), 3840x2160 [SAR 1:1 DAR 16:9], 5164 kb/s, 25 fps, 25 tbr, 12800 tbn, 25 tbc (default)
    Metadata:
      handler_name    : VideoHandler
      vendor_id       : [0][0][0][0]
  Stream #0:1(jpn): Audio: aac (HE-AAC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 62 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
Stream mapping:
  Stream #0:0 -> #0:0 (hevc (native) -> h264 (h264_vaapi))
  Stream #0:1 -> #0:1 (aac (native) -> aac (native))
Press [q] to stop, [?] for help
[h264_vaapi @ 0x55571c6a2200] Driver does not support VBR RC mode (supported modes: CQP).
Error initializing output stream 0:0 -- Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
[aac @ 0x55571c750140] Qavg: 50571.980
[aac @ 0x55571c750140] 2 frames left in the queue on closing
Conversion failed!

**Reply by nyanmisaka** [reply]
https://jellyfin.org/docs/general/administration/hardware-acceleration.html#intel-gen9-and-gen11-igpus

Intel Guc/Huc is required to be enabled on this Jasper Lake chip but your QNAP NAS is lacking of the necessary firmware. Since normal user is not allowed to modify the system partition of the QNAP QTS system. Contact your device provider to solve this issue.

You can get the error message by running `sudo dmesg | grep i915` in ssh and attach them to QNAP customer service.

**Reply by zhongjiongsheng** [reply]
OK! So it seems to be a QTS system issue. THX A LOT! I will contact the provider.

**Reply by nyanmisaka** [reply]
Just let you know that your N4505 is capable of doing VAAPI and QSV transcoding once QNAP fixes this issue in the future updates.

Two prerequisites:
1. The kernel paramter `i915.enable_guc=2` must be set. <= Achieved
2. These files must be available in the booting stage. <= Lacking of firmware
```
/lib/firmware/i915/icl_dmc_ver1_09.bin
/lib/firmware/i915/ehl_guc_33.0.4.bin
```

The error message should be print like this:
```
[    4.825326] i915 0000:00:02.0: Direct firmware load for i915/icl_dmc_ver1_09.bin failed with error -2
[    4.828269] i915 0000:00:02.0: Direct firmware load for i915/ehl_guc_33.0.4.bin failed with error -2
[    4.835637] i915 0000:00:02.0: Falling back to sysfs fallback for: i915/icl_dmc_ver1_09.bin
[    4.845837] i915 0000:00:02.0: Falling back to sysfs fallback for: i915/ehl_guc_33.0.4.bin
[   66.748942] i915 0000:00:02.0: [drm] Failed to load DMC firmware i915/icl_dmc_ver1_09.bin. Disabling runtime power management.
[   66.748950] i915 0000:00:02.0: [drm] GuC firmware i915/ehl_guc_33.0.4.bin: fetch failed with error -110
[   66.761687] i915 0000:00:02.0: [drm] DMC firmware homepage: https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/i915
[   66.786761] i915 0000:00:02.0: [drm] GuC firmware(s) can be downloaded from https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/i915
[   66.803472] i915 0000:00:02.0: [drm] GuC is uninitialized
```
